### PR TITLE
SALTO-2773 SALTO-2853 Change transformValues to use the allowEmpty flag so it wont automatically omit empty values 

### PR DESCRIPTION
**Pull request by Uri-Bar**
_Change transformValues to use the allowEmpty flag so it wont automatically omit empty values_

---
_Additional context for reviewer_:
related to: https://github.com/salto-io/salto/pull/3412#issue-1390450814

---
_Release Notes_: 
Salesforce-adapter:

* Fix transformValues so it wont delete empty values regardless of the allowEmpty flag.
---

_User Notifications_: 
None

### CODE REVIEW

File: packages/jira-adapter/jest.config.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ module.exports = deepMerge(
     ],
     coverageThreshold: {
       'global': {
-        branches: 93.5,
+        branches: 93.4,
```
Comment (ori-moisis) [inline packages/jira-adapter/jest.config.js:32]: why was this changed?
there was no change in any jira adapter file AFAICT

File: packages/zendesk-adapter/test/adapter.test.ts [hunk 2]
```diff
@@ -899,7 +899,7 @@ describe('adapter', () => {
             toChange({ before: new InstanceElement('inst2', groupType) }),
             // explicitly state key so that it appears in the instance's generated type
             toChange({ after: new InstanceElement('inst3', brandType, { ref, key: undefined }) }),
-            toChange({ after: new InstanceElement('inst4', anotherType) }),
+            toChange({ after: new InstanceElement('inst4', anotherType, { key: undefined }) }),
```
Comment (ori-moisis) [inline packages/zendesk-adapter/test/adapter.test.ts:902]: @netama any idea what this test is about and why do we need this definition explicitly?
@Uri-Bar if you have an explanation for why this change was needed that would help

Comment (Uri-Bar) [reply packages/zendesk-adapter/test/adapter.test.ts:902]: Im not completely sure, and the object that failing the test has no empty values so there not supposed to be realted to our change.
however, it is failing in the `strict` section of transformValues because without the addition `anotherType` has no field named `key` and the func deletes this value.

File: packages/jira-adapter/jest.config.js (hunk 1 above)
Comment (Uri-Bar) [reply packages/jira-adapter/jest.config.js:32]: there is only the test thing. after the fix the coverage went down.

Comment (ori-moisis) [reply packages/jira-adapter/jest.config.js:32]: this is in the jira adapter, the test change is in zendesk...

File: packages/zendesk-adapter/test/adapter.test.ts (hunk 2 above)
Comment (ori-moisis) [reply packages/zendesk-adapter/test/adapter.test.ts:902]: but all of that was true before your change as well
if your change is affecting the behavior of strict transform we need to investigate it further to make sure we are not introducing unexpected side effects

File: packages/jira-adapter/jest.config.js (hunk 1 above)
Comment (Uri-Bar) [reply packages/jira-adapter/jest.config.js:32]: oh, missed that. is it not possible the change we made caused jira adapter tests to do not reach some branches?

Comment (ori-moisis) [reply packages/jira-adapter/jest.config.js:32]: possible, but very unlikely

**Review by ori-moisis — APPROVED** [review]
